server/src/routes/dto/expenses.rs

```diff
@@ -0,0 +1,24 @@
+#![allow(missing_docs)]
+
+use entity::{Id, MoneyAmount};
+use serde::Deserialize;
+use validator::Validate;
+
+#[derive(Deserialize, Validate)]
+#[serde(rename_all = "camelCase")]
+pub struct NewExpenseRequest {
+    #[validate(length(min = 1, max = 255, message = "expense name must not be empty"))]
+    pub name: String,
+
+    pub description: String,
+
+    pub recurrence_type_id: Id,
+
+    pub currency_type_id: Id,
+
+    pub predefined_expense_id: Option<Id>,
+
+    pub start_date: String,
+
+    pub value: MoneyAmount,
```

Comment:
Validations are missing for the length of the description and value. We have a database lengths set for this decimal so methinks we should constrain it. 
![image](https://user-images.githubusercontent.com/37137309/193069181-369cfadc-3e37-42be-9462-7b82d5bf255c.png)
![image](https://user-images.githubusercontent.com/37137309/193069091-d647ff41-4c97-4a71-acb7-bd284c2b80d3.png)

Reply:
Unfortunately we cannot validate _value_ as it is not recognized neither as a string or a number.
So validator says (the message is similar with range): 
```
Validator `length` can only be used on types `String`, `&str`, Cow<'_,str>, `Vec`, slice, or map/set types (BTree/Hash/Index) but found `MoneyAmount` for field `value` 
```

Description validation is fixed in https://github.com/szattila98/veryrezsi/pull/40/commits/0f1ecec443ef4d0610411f276899e3ae507e2013

Reply:
Do i smell a custom validator? 
I will include it in my testing stuff.